app/models/repaymentHistory/RepaymentHistoryUtils.scala

```diff
@@ -62,7 +63,7 @@ object RepaymentHistoryUtils {
 
   def getGroupedPaymentHistoryData(payments: List[Payment], repayments: List[RepaymentHistory], isAgent: Boolean,
                                    MFACreditsEnabled: Boolean, CutOverCreditsEnabled: Boolean, languageUtils: LanguageUtils
-                                  )(implicit messages: Messages): List[(Int, List[PaymentHistoryEntry])] = {
+                                  )(implicit messages: Messages, dateService: DateService): List[(Int, List[PaymentHistoryEntry])] = {
```

Comment:
Please use DateServiceInterface instead, this would give us option to switch actual implementation under test context